src/Pages/Detail/Detail.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import API from '../../config';
+import Goods from './DetailSections/Goods';
+import Categories from './DetailSections/Categories';
+import Popup from '../../Components/ShoppingModal/Popup';
+import Login from '../Login/Login';
+import './Detail.scss';
+
+class Detail extends React.Component {
+  state = {
+    goods: {},
+    isModalOpen: false,
+    isLogin: false,
+    content: {},
+  };
+
+  handleModal = (e, amount) => {
+    const { name } = e.target;
+    const { isLogin, goods } = this.state;
+    const { id, korean_name, price } = goods;
+
+    const order = { id, korean_name, amount, price };
+
+    if (!isLogin) {
+      this.addOrder(order);
```

Comment:
비동기 상태를 다루기 위해 자바스크립트에서는 `Promise` 객체를 사용합니다. 자세한 구현사항을 지금 설명할 필요는 없을 것 같고 다음과 같이 사용해주세요
```suggestion
      this.addOrder(order)
        .then(res => ...)
```
```suggestion
const 
  handleModal = async (e, amount) => {
  
  await this.addOrder(order)
  ...
```

Reply:
아래의 방법으로 처리하였습니다.